src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,90 @@
+package duke;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents the storage for the Duke application.
+ * Handles the loading and saving of tasks to a file.
+ */
+public class Storage {
+    private String filePath;
+
+    /**
+     * Constructs a new Storage with the specified file path.
+     *
+     * @param filePath The path of the file to load from and save to.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads tasks from the file.
+     *
+     * @return A list of tasks loaded from the file.
+     * @throws DukeException If there's an error loading from the file.
+     */
+    public List<Task> load() throws DukeException {
+        List<Task> tasks = new ArrayList<>();
+        Path path = Paths.get(filePath);
+        if (!Files.exists(path.getParent())) {
+            try {
+                Files.createDirectories(path.getParent());
+            } catch (Exception e) {
+                e.printStackTrace();
```

Comment:
Would it be beneficial to handle the exception more gracefully by throwing a custom exception, like 'DukeException', or displaying a more user friendly message?

e.g.
```java
try {
    Files.createDirectories(path.getParent());
} catch (IOException e) {
    System.err.println("An I/O error occurred while creating directories: " + e.getMessage());
} catch (SecurityException e) {
    System.err.println("A security error occurred while creating directories: " + e.getMessage());
}
```